website/docs/reference/administrative-commands/createIndexes.mdx

```diff
@@ -0,0 +1,152 @@
+| `createIndexes`           | string   | The name of the collection.                                                                                                                     |
+| `indexes`                 | array    | An array of index specification documents. Each document defines an index.                                                                      |
+| `key`                     | document | A single document that contains the field(s) to index and their sort order or type (`1`, `-1`, `text`, `cosmosSearch` for vector indexes, etc). |
+| `name`                    | string   | A required name for the index. Must be unique to the collection.                                                                                |
+| `unique`                  | boolean  | Optional. Ensures all values are unique for the indexed field if set to `true`.                                                                 |
+| `partialFilterExpression` | document | Optional. Restricts index to documents matching the condition.                                                                                  |
+| `sparse`                  | boolean  | Optional. Only indexes documents that contain the indexed field if set to `true`.                                                               |
+
+## Example usage
+
+The following examples demonstrate how to create different types of indexes.
+
+### Simple index
+
+The simplest way to create an index is to use the `createIndexes` command with the `key` field (the field(s) to be indexed) and their sort order (`1` for ascending, `-1` for descending).
+
+This example creates a simple index on the `title` field in the `books` collection.
+
+<CodeBlock language="js">{CreateIndex}</CodeBlock>
+
+### Compound index
+
+Compound indexes enable multiple fields to be indexed together.
+
+The example below creates a compound index on the `publication.date` and `analytics.average_rating` fields in the `books` collection.
+
+<CodeBlock language="js">{CreateCompoundIndex}</CodeBlock>
+
+### Unique index
+
+Unique indexes ensure that your indexed fields only contain unique values, without duplicates.
+
+This example creates a unique index on the `isbn` field in the `books` collection to ensure that each book has a unique ISBN.
+
+<CodeBlock language="js">{CreateUniqueIndex}</CodeBlock>
+
+### Partial index
+
+A partial index is similar to a regular index, however, it only indexes a fraction of documents in a collection based on a specified filter condition.
+
+Let's create a partial index on the `books` collection only for books published after the year 2000.
+
+<CodeBlock language="js">{CreatePartialIndex}</CodeBlock>
+
+### Sparse index
+
+Sparse indexes only apply on documents with the specified index field. Documents without the indexed field are not included in the index.
```

Comment:
Let me know if my understanding of sparse index is correct? If a document doesn't contain field `x` and an index without `sparse` is created, a document is indexed on field `x`, but since the field doesn't exist, it uses `null` for indexing. In bson, `null` is equivalent of non existent field.
If `sparse` index is created, a document with non-existent field `x` is skipped, just simply won't be indexed.

You are trying to describe the same or is my understanding not quite correct? I just feel the current sentences don't quite give full description. :thinking:

Reply:
Sparse only indexes the documents that contain the specified field. If the field doesn't exist in that documents, it's not indexed.

Reply:
On the syntax, you were right with one of your earlier comments. Having them compact like that is not proper since some of the indexes cannot be specified together. With separated index syntaxes like it is now, that's actually a valid JS and could even run in mongosh.